WarOOP/Models/Army.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using System.Threading.Channels;
+using WarOOP.Abstractions;
+
+namespace WarOOP.Models;
+
+public class Army : IArmy
+{
+    private List<Warrior> Units;
```

Comment:
I would add _readonly_ modifier. I don't think that you will need to reassign the collection